src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -82,6 +82,21 @@ export interface WrappedResourceModel {
  * The interface that resource resolvers must implement
  */
 export interface BranchDataProvider<TResource extends ResourceBase, TModel extends ResourceModelBase> extends vscode.TreeDataProvider<TModel> {
+    /**
+     * @deprecated This is implemented by the Resources API and does not need to be implemented by the
+     * {@link BranchDataProvider}.
+     */
+    getParent?: never;
+
+    /**
+     * Get the children of `element`. This differs from standard {@link vscode.TreeDataProvider}
+     * in that the `element` parameter will never be undefined.
+     *
+     * @param element The element from which the provider gets children. Cannot be `undefined`.
+     * @return Children of `element`.
+     */
+    getChildren(element: TModel): TModel[] | Thenable<TModel[]>;
```

Comment:
In line with my suggestion, we can override with this `getChildren(element: TModel)` to indicate that `element` will never be `undefined`.